src/Microsoft.AspNet.OData.Shared/EdmChangedObjectCollection.cs

```diff
@@ -57,5 +69,288 @@ private void Initialize(IEdmEntityType entityType)
             _edmType = new EdmDeltaCollectionType(new EdmEntityTypeReference(_entityType, isNullable: true));
             _edmTypeReference = new EdmCollectionTypeReference(_edmType);
         }
+
+        /// <summary>
+        /// Patch for Types without underlying CLR types
+        /// </summary>
+        /// <param name="originalCollection"></param>
+        /// <returns>ChangedObjectCollection response</returns>
+        public EdmChangedObjectCollection Patch(ICollection<EdmStructuredObject> originalCollection)
+        {            
+            PatchHandler = new DefaultTypelessPatchHandler(originalCollection, _entityType);
+
+            return CopyChangedValues();
+        }
+
+        /// <summary>
+        /// Handler for users Create, Get and Delete Methods
+        /// </summary>
+        internal TypelessPatchMethodHandler PatchHandler { get; set; }
+
+        /// <summary>
+        /// Patch for EdmChangedObjectCollection, a collection for IEdmChangedObject 
+        /// </summary>
+        /// <returns>ChangedObjectCollection response</returns>
+        public EdmChangedObjectCollection Patch(TypelessPatchMethodHandler patchHandler)
+        {
+            PatchHandler = patchHandler;
+            return CopyChangedValues();
+        }
+
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1800:DoNotCastUnnecessarily")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Maintainability", "CA1506:AvoidExcessiveClassCoupling")]
+        internal EdmChangedObjectCollection CopyChangedValues()
+        {
+            EdmChangedObjectCollection changedObjectCollection = new EdmChangedObjectCollection(_entityType);
+            List<IEdmStructuralProperty> keys = _entityType.Key().ToList();
```

Comment:
>List<IEdmStructuralProperty> keys [](start = 12, length = 33)

Since we never add/remove from this list, probably more efficient to make this an IEdmStructuralProperty[] rather than List<IEdmStructuralProperty>.